### PR TITLE
Fixes #1631 - Warns user and aborts tests if server isn't right or if it's not in test mode.

### LINKED ISSUE #1631: [tests] Starting "python run.py -t" from the functional tests
So we have a pattern. 

Very often in the team, we forget (or at least I do) to start `python run.py -t` for running the functional tests. And indeed we have quite a lot of things to do before starting the tests. Having the right version of java, starting the server, the long command for launching the tests.

I wonder if this should not be handled with a script.

### PR DESCRIPTION
**Pull request by brizental**
So, to check that the server is in test mode, I make a request to our API and check that it's returning fixture data. This took a little while longer, because I had to find a workaround to add external modules to the `intern.js` config file, since you can't add them as you would on the test suites, because it's not loaded as an AMD module. I don't feel that my solution looks very good, but it works and I asked in the intern Gitter chat and the guys said it was a fine way to do it.

Anyways, I also removed code to login with real data to GitHub. I know that @miketaylr was kind of against it, but I found out that we actually removed the code that did that in #1604 , so it didn't work anyways... I think we could just open a new issue for that if you guys think it's necessary to have the option to test without fixture data. 

I also removed the `visibleByClass` method, because it wasn't used anywhere and Leadfoot has a similar method ([findDisplayedByClassName](https://theintern.io/leadfoot/module-leadfoot_Command.html#findDisplayedByClassName)).

Lastly, I noticed that when you start the server in non-test mode it runs grunt before starting the server, but not in test mode. So I fixed that.

Hope it's alright to add all this to the same PR. I didn't think it was necessary to open new issues for these things.

r? @miketaylr @zoepage

### CODE REVIEW

**Comment by zoepage** [comment]
@brizental great work :) there is an issue somewhere mentioning the geckobug. Could you reference this in the commit? (Sorry, on my phone right now which makes it hard to search for it).

Thanks!

**Comment by brizental** [comment]
@zoepage #1590

File: tests/functional/lib/helpers.js [hunk 1]
```diff
@@ -49,90 +44,68 @@ define(
...
+          if (response.headers.hasOwnProperty("etag")) {
+            reject(
+              new Error(
+                `
+              =======================================================
+              It seems like you didn't start the server in test mode.
+              Open another terminal and window type: 
+             \x1b[32m npm run start:test\x1b[0m
+              or
+             \x1b[32m source env/bin/activate && python run.py -t\x1b[0m
```
Comment (miketaylr) [inline tests/functional/lib/helpers.js]: Can we have this second line just mention `python run.py -t`? The first does effectively the same, but assumes you used `virtualenv` in a specific way. This second way is more general.

File: tests/functional/lib/helpers.js [hunk 2]
```diff
@@ -49,90 +44,68 @@ define(
...
+    /* 
+    This method makes a call to our API to check that the server is returning fixture data,
+    it will also check if there's anything wrong with the server.
+  */
+    function checkServer() {
+      return new Promise(function(resolve, reject) {
+        var request = http.get(url("/api/issues/100"), function(response) {
+          // If the headers are from non-fixture data,
+          // they will have the "etag" property.
+          if (response.headers.hasOwnProperty("etag")) {
```
Comment (miketaylr) [inline tests/functional/lib/helpers.js]: Another way would be to look for the `_fixture` property on the first object in the array:

https://github.com/webcompat/webcompat.com/blob/03b0da41a8615f160a90529fef51dea64352d0c6/tests/fixtures/api/issues/100/comments.json#L3

(pretty sure I added that to all fixtures manually... )

File: tests/functional/lib/helpers.js [hunk 3]
```diff
@@ -49,90 +44,68 @@ define(
       );
     }
 
-    // TODO: when https://github.com/mozilla/geckodriver/issues/308 is fixed,
-    // remove this ugliness.
-
-    function login(context) {
```
Comment (miketaylr) [inline tests/functional/lib/helpers.js]: Can you file a follow up issue to create a way to test logging in, please?

File: tests/functional/lib/helpers.js [hunk 4]
```diff
@@ -49,90 +44,68 @@ define(
...
+        request.on("error", function() {
+          reject(
+            new Error(
+              `
+            ======================================================
+            Oops, something went wrong. Try restarting the server.
+            Open another terminal and window type: 
+           \x1b[32m npm run start:test\x1b[0m
+            or
+           \x1b[32m source env/bin/activate && python run.py -t\x1b[0m
```
Comment (miketaylr) [inline tests/functional/lib/helpers.js]: Same here.

File: tests/intern.js [hunk 5]
```diff
@@ -25,17 +19,28 @@ define(["intern"], function(intern, topic) {
...
+    },
+
     // The port on which the instrumenting proxy will listen
     proxyPort: 9090,
 
     // A fully qualified URL to the Intern proxy
     proxyUrl: "http://127.0.0.1:9090/",
     siteRoot: siteRoot,
     tunnel: "SeleniumTunnel",
-    tunnelOptions: {
```
Comment (miketaylr) [inline tests/intern.js]: Can you explain this part to me? Why don't we want to specify the drivers?

File: tests/functional/lib/helpers.js (hunk 2 above)
Comment (brizental) [reply tests/functional/lib/helpers.js]: That's cool! I didn't know that. I'll change it :)

File: tests/functional/lib/helpers.js (hunk 3 above)
Comment (brizental) [reply tests/functional/lib/helpers.js]: I think we already have and issues like that, wait...
Yeah! #1385

File: tests/intern.js (hunk 5 above)
Comment (brizental) [reply tests/intern.js]: That was a mistake. I removed it while testing something, but should have put it back. Thanks for noticing.

File: tests/functional/lib/helpers.js (hunk 1 above)
Comment (brizental) [reply tests/functional/lib/helpers.js]: Yeah, sure. Makes sense.

Comment (miketaylr) [reply tests/functional/lib/helpers.js]: Thanks!

File: tests/functional/lib/helpers.js (hunk 3 above)
Comment (miketaylr) [reply tests/functional/lib/helpers.js]: Ah, perfect. Thanks!

**Comment by brizental** [comment]
@miketaylr I made the changes you asked for, Mike. Had to squash all commits together because I did something stupid while rebasing the first time, lol sorry.

**Comment by miketaylr** [comment]
> Had to squash all commits together because I did something stupid while rebasing the first time, lol sorry.

Heh no worries. Been there!

**Comment by miketaylr** [comment]
You rock, Bea -- thanks!

**Pull request merged**